app/backend/src/couchers/models.py

```diff
@@ -2667,3 +2673,37 @@ class AccountDeletionReason(Base):
     reason = Column(String, nullable=True)
 
     user = relationship("User")
+
+
+class UserGroupType(enum.Enum):
+    duplicate_account = enum.auto()
+
+
+class UserGroup(Base):
+    """
+    Represents a group of users grouped by a specific type
+    """
+
+    __tablename__ = "user_groups"
+
+    id = Column(BigInteger, primary_key=True)
+    group_type = Column(Enum(UserGroupType), nullable=False)
+    created = Column(DateTime(timezone=True), nullable=False, server_default=func.now())
+
+    # Relationships
+    users = relationship("User", secondary="user_group_members", back_populates="groups")
+
+    def has_user(self, user):
+        """Check if user is in group"""
+        return user.id in {u.id for u in self.users}
```

Comment:
This kind of check is better done in SQL. Right now when you call this function, it will pull the set of users for this group into python, construct the set, then check membership. With a SQL index, you can do that in very little computation on the database side.